impl/client/src/main/javascript/web/util/URLEncoder.js

```diff
@@ -70,22 +78,24 @@ define("common-ui/util/URLEncoder", [], function() {
       args = [args];
     }
 
-    // Detect the presence of the "?" to determine
-    // when the special double-slash encoding should end.
-    var pathPart = str.split("?")[0];
-    var pathBounds = (pathPart.match(/\{[\d]+\}/g) || []).length;
+    if (doubleEncodeSlashes) {
+      // Detect the presence of the "?" to determine
+      // when the special double-slash encoding should end.
+      var pathPart = str.split("?")[0];
+      var pathBounds = (pathPart.match(/\{[\d]+\}/g) || []).length;
 
-    args = (args || []).map(function(item, pos) {
+      args = (args || []).map(function (item, pos) {
 
-      var encodedStr = encodeURIComponent(String(item));
+        var encodedStr = encodeURIComponent(String(item));
 
-      // Double-encode / and \ to work around Tomcat issue.
-      if(pos < pathBounds) {
-        encodedStr = encodedStr.replace(/%5C/g, "%255C").replace(/%2F/g, "%252F");
-      }
+        // Double-encode / and \ to work around Tomcat issue.
+        if (pos < pathBounds) {
+          encodedStr = encodedStr.replace(/%5C/g, "%255C").replace(/%2F/g, "%252F");
+        }
 
-      return encodedStr;
-    });
+        return encodedStr;
+      });
+    }
```

Comment:
update `URLEncoder.spec.js`

you previous made changes to https://github.com/pentaho/pentaho-platform-plugin-common-ui/blob/master/impl/client/src/test/javascript/util/URLEncoder.spec.js

If you had to make changes to the source code, then shouldn't you have to make changes to the automated test code as well?